### PR TITLE
refactor: updated examples folder structure

### PR DESCRIPTION
**Pull request by raresgaia123**
created folders for each example
necessary for documenting each example

## Description

Please provide a meaningful description of what this change will do, or is for. Bonus points for including links to
related issues, other PRs, or technical references.

Note that by _not_ including a description, you are asking reviewers to do extra work to understand the context of this
change, which may lead to your PR taking much longer to review, or result in it not being reviewed at all.

## Type of Change

- [ ] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [x] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [ ] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

**Comment by raresgaia123** [comment]
Converted to draft untill: https://github.com/cisco-open/pymultiworld/pull/36 is merged.

File: examples/all_reduce/multiworld_all_reduce.py
Comment (myungjin) [inline examples/all_reduce/multiworld_all_reduce.py:1]: rename it to m8d.py

File: examples/resnet/resnet_multiworld.py
Comment (myungjin) [inline examples/resnet/resnet_multiworld.py:1]: rename it to m8d.py.

File: examples/send_recv/m8d_asyncio.py
Comment (myungjin) [inline examples/send_recv/m8d_asyncio.py:1]: rename it to m8d.py

File: examples/single_world/single_world.py
Comment (myungjin) [inline examples/single_world/single_world.py:1]: move it into send_recv folder and rename it to single_world.py.

**Review by myungjin — COMMENTED** [review]
left comments

File: README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ m8d-post-setup
 
 ## Running Examples
 
-* [`multiworld_asyncio.py`](/examples/multiworld_asyncio.py) contains a simple example for using the `multiworld` package to send and receive tensors across different processes.
+* [`m8d_send_recv.py`](/examples/send_recv/m8d_send_recv.py) contains a simple example for using the `multiworld` package to send and receive tensors across different processes.
```
Comment (myungjin) [inline README.md:75]: it's m8d.py, not m8d_send_recv.py

File: README.md [hunk 2]
```diff
@@ -87,11 +87,11 @@ For example, run the following commands, by changing the IP address (10.20.1.50)
 
 ```bash
 # on terminal window 1
-python multiworld_asyncio.py --backend nccl --worldinfo 1,0 --worldinfo 2,0 --addr 10.20.1.50
+python m8d_send_recv.py --backend nccl --worldinfo 1,0 --worldinfo 2,0 --addr 10.20.1.50
```
Comment (myungjin) [inline README.md:90]: m8d.py

File: examples/all_reduce/m8d.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 """
-multiworld_all_reduce.py: This script demonstrates how to
+m8d_all_reduce.py: This script demonstrates how to
```
Comment (myungjin) [inline examples/all_reduce/m8d.py:18]: m8d.py

File: examples/resnet/m8d.py [hunk 4]
```diff
@@ -15,16 +15,16 @@
 # SPDX-License-Identifier: Apache-2.0
 
 """
-resnet_multiworld.py: This script demonstrates how to run a ResNet model on multiple worlds using PyTorch.
+m8d_resnet.py: This script demonstrates how to run a ResNet model on multiple worlds using PyTorch.
```
Comment (myungjin) [inline examples/resnet/m8d.py:18]: m8d.py

File: examples/resnet/m8d.py [hunk 5]
```diff
@@ -15,16 +15,16 @@
...
 
 Summary:
     - The script initializes a ResNet model on every worker.
     - The leader sends an image to a worker for inference.
     - The worker processes the image and sends the predicted class back to the leader.
 
 Sample usage:
-    Single host: python resnet_multiworld.py --num_workers 1 --backend gloo
-    Multi host: python resnet_multiworld.py --num_workers 2 --backend nccl --multihost --addr 10.20.1.50 --rank 0
+    Single host: python m8d_resnet.py --num_workers 1 --backend gloo
```
Comment (myungjin) [inline examples/resnet/m8d.py:26]: m8d.py

File: examples/send_recv/m8d.py [hunk 6]
```diff
@@ -15,7 +15,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
 """
-multiworld_asyncio.py: This script is a modified version of examples/leader_recv.py.
+m8d_send_recv.py: This script is a modified version of examples/leader_recv.py.
```
Comment (myungjin) [inline examples/send_recv/m8d.py:18]: m8d.py

**Review by myungjin — COMMENTED** [review]
left comments

**Review by myungjin — APPROVED** [review]
lgtm

**Pull request merged**